echopop/statistics.py

```diff
@@ -0,0 +1,268 @@
+
+    # ---- Overall total using the overall mean
+    replicate_total = total_arr.sum(axis=1)
+
+    # Output the related summary statistics
+    # ---- Save the output resampled distributions
+    resampled_distributions = pd.DataFrame(
+        {
+            "realization": np.arange(1, transect_replicates + 1),
+            "mean_standardized": replicate_mean,
+            "standard_deviation": replicate_stdev,
+            "variance": replicate_variance,
+            "cv": replicate_cv,
+            "mean": replicate_sum_mean,
+            "total": replicate_total,
+        }
+    )
+    # ---- Save the stratified results
+    stratified_results = {
+        "variable": settings_dict["variable"],
+        "ci_percentile": 0.95,
+        "num_transects": strata_summary["transect_count"].sum(),
+        "total_area": strata_summary["transect_area_total"].sum(),
+        "mean": {
+            "weighted_estimate": np.mean(replicate_mean),
+            "unweighted_estimate": np.mean(replicate_sum_mean),
+            "weighted_confidence_interval": confidence_interval(replicate_mean),
+            "unweighted_confidence_interval": confidence_interval(replicate_sum_mean),
+        },
+        "variance": {
+            "estimate": np.mean(replicate_variance),
+            "confidence_interval": confidence_interval(replicate_variance),
+        },
+        "cv": {
+            "estimate": np.mean(replicate_cv),
+            "confidence_interval": confidence_interval(replicate_cv),
+        },
+        "total": {
+            "estimate": np.mean(replicate_total),
+            "confidence_interval": confidence_interval(replicate_total),
+        },
+    }
+    # ---- Return outputs
+    return resampled_distributions, stratified_results
+
+
+def array_math(left_array: np.ndarray, right_array: np.ndarray, operation: str) -> np.ndarray:
+    """
+    Helper function that applies arithmetric operations across the dimensions of two arrays.
+
+    Parameters
+    ----------
+    left_array: np.ndarray
+        The first array containing data.
+    right_array: np.ndarray
+        The second array containing data.
+    operation: str
+        A string value representing a basic arithmetic operation ('+', '+', '-', '/')
+    """
+    # Define operations dictionary key
+    operations = {"+": np.add, "-": np.subtract, "*": np.multiply, "/": np.divide}
+
+    # Apply the operation and return the output
+    return operations[operation](left_array.transpose(), right_array).transpose()
+
+
+def confidence_interval(values: np.ndarray) -> np.ndarray:
+    """
+    Calculates the 95% confidence interval (Normal) for a given array
+
+    Parameters
+    ----------
+    values: np.ndarray
+        An array of values
+
+    Notes
+    -----
+    This function calculates the 95% confidence interval (assuming a Normal) distribution
+    for the bootstrapped stratified samples. This is done as opposed to using the percentile
+    method for estimate the intervals.
```

Comment:
I wonder why we can assume normal here? Is that the underlying assumption for what the stratified distributions would follow?

Reply:
Mostly for the sake of simplicity since the default number of replicates is 10,000, and typically it's assumed that (boostrapped) stratified samples will be asymptotic Gaussian although that is not always the case (nor a required assumption). That being said, I have implemented a few additional parameteric and non-parametric methods for computing the CI's (which will be useful way down the road when we formally incorporate bootstrapping at a later release). The augmented code now  defaults to the Bias-Corrected and Accelerated (BCa) non-parameteric method to compute the bootstrap intervals (distribution skewness is estimated via jackknife to compute the acceleration parameter). In cases where the distributions are too skewed and therefore outside the scope of the BCa, the implementation reverts to using the studentized intervals with jackknife bias-correction.